From 87e355494fbe83602718c80082f8a96e0415e2e0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 06 Sep 2020 11:13:39 +0300
Subject: [PATCH] [SW-0] boot_if: move module to common

The boot_if module should be used by all entities communicating with the
host S/W. This includes the ppboot, preboot, and recovery.
Move it to common space.

Change-Id: I27a2779a5dad8f5b1abefab6460c186ab7d22dad
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/boot_if.c b/common/boot_if.c
new file mode 100644
index 0000000..6d848ace
--- /dev/null
+++ b/common/boot_if.c
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright 2016-2020 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#include <stdbool.h>
+
+#include <psoc_global_conf_regs.h>
+
+#include <hl_boot_if.h>
+#include <preboot.h>
+
+/*
+ * The cpu_boot_cmd_clr_before_wfe() function should NOT be used in general as
+ * the LKD message register should be written by LKD solely.
+ *
+ * There is only one special case when this function can be used is only when
+ * the LKD performs hard reset, which may mean the H/W is not reliable and
+ * writing to PCI might not complete and result in the register not cleared.
+ * So to be on the safe side, when receiving the WFE, clear the message register
+ */
+void cpu_boot_cmd_clr_before_wfe(void)
+{
+	regmap_writel(KMD_MSG_NA, mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU);
+}
+
+enum kmd_msg cpu_boot_cmd_get(void)
+{
+	return regmap_readl(mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU);
+}
+
+void cpu_boot_sts_set(u32 val)
+{
+	regmap_writel(val, mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS);
+}
+
+/* We support only ERR0 for now */
+void cpu_boot_err_clrset(u32 mask, bool set)
+{
+	u32 val = regmap_readl(mmCPU_BOOT_ERR0) & ~mask;
+
+	if (set)
+		val |= mask;
+
+	regmap_writel(val, mmCPU_BOOT_ERR0);
+}
+
+void cpu_boot_rsp_set(enum cpu_msg_status sts)
+{
+	regmap_writel(sts, mmCPU_CMD_STATUS_TO_HOST);
+}
diff --git a/include/boot_if.h b/include/boot_if.h
new file mode 100644
index 0000000..e0f54c0
--- /dev/null
+++ b/include/boot_if.h
@@ -0,0 +1,18 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * Copyright 2016-2020 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#ifndef _HL_BOOT_IF_H_
+#define _HL_BOOT_IF_H_
+
+#include <hl_boot_if.h>
+
+enum kmd_msg cpu_boot_cmd_get(void);
+void cpu_boot_cmd_clr_before_wfe(void);
+void cpu_boot_sts_set(u32 val);
+void cpu_boot_err_clrset(u32 mask, bool set);
+void cpu_boot_rsp_set(enum cpu_msg_status sts);
+
+#endif /* _HL_BOOT_IF_H_ */
diff --git a/preboot/boot_if.c b/preboot/boot_if.c
deleted file mode 100644
index 6d848ace..0000000
--- a/preboot/boot_if.c
+++ /dev/null
@@ -1,52 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0+
-/*
- * Copyright 2016-2020 HabanaLabs, Ltd.
- * All Rights Reserved.
- */
-
-#include <stdbool.h>
-
-#include <psoc_global_conf_regs.h>
-
-#include <hl_boot_if.h>
-#include <preboot.h>
-
-/*
- * The cpu_boot_cmd_clr_before_wfe() function should NOT be used in general as
- * the LKD message register should be written by LKD solely.
- *
- * There is only one special case when this function can be used is only when
- * the LKD performs hard reset, which may mean the H/W is not reliable and
- * writing to PCI might not complete and result in the register not cleared.
- * So to be on the safe side, when receiving the WFE, clear the message register
- */
-void cpu_boot_cmd_clr_before_wfe(void)
-{
-	regmap_writel(KMD_MSG_NA, mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU);
-}
-
-enum kmd_msg cpu_boot_cmd_get(void)
-{
-	return regmap_readl(mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU);
-}
-
-void cpu_boot_sts_set(u32 val)
-{
-	regmap_writel(val, mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS);
-}
-
-/* We support only ERR0 for now */
-void cpu_boot_err_clrset(u32 mask, bool set)
-{
-	u32 val = regmap_readl(mmCPU_BOOT_ERR0) & ~mask;
-
-	if (set)
-		val |= mask;
-
-	regmap_writel(val, mmCPU_BOOT_ERR0);
-}
-
-void cpu_boot_rsp_set(enum cpu_msg_status sts)
-{
-	regmap_writel(sts, mmCPU_CMD_STATUS_TO_HOST);
-}
diff --git a/preboot/boot_if.h b/preboot/boot_if.h
deleted file mode 100644
index e0f54c0..0000000
--- a/preboot/boot_if.h
+++ /dev/null
@@ -1,18 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0+ */
-/*
- * Copyright 2016-2020 HabanaLabs, Ltd.
- * All Rights Reserved.
- */
-
-#ifndef _HL_BOOT_IF_H_
-#define _HL_BOOT_IF_H_
-
-#include <hl_boot_if.h>
-
-enum kmd_msg cpu_boot_cmd_get(void);
-void cpu_boot_cmd_clr_before_wfe(void);
-void cpu_boot_sts_set(u32 val);
-void cpu_boot_err_clrset(u32 mask, bool set);
-void cpu_boot_rsp_set(enum cpu_msg_status sts);
-
-#endif /* _HL_BOOT_IF_H_ */
diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index b4e3836..95fa879 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -7,10 +7,10 @@
 #include <string.h>
 #include <hl_boot_if.h>
 #include <preboot.h>
+#include <boot_if.h>
 
 #include "cmd_module.h"
 #include "uart.h"
-#include "boot_if.h"
 #include "fit.h"
 
 struct lkd_status {
diff --git a/preboot/preboot.c b/preboot/preboot.c
index 3f95bf1..27d4178 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -12,11 +12,11 @@
 #include <preboot.h>
 #include <hl_fw_if.h>
 #include <hl_boot_if.h>
+#include <boot_if.h>
 
 #include "common.h"
 #include "uart.h"
 #include "pll.h"
-#include "boot_if.h"
 #include "cmd_module.h"
 
 static struct cmd_module_ops cmd_ops[] = {
